core/commands/ipns.go

```diff
@@ -57,9 +59,10 @@ Resolve the value of a dnslink:
 	Options: []cmdkit.Option{
 		cmdkit.BoolOption("recursive", "r", "Resolve until the result is not an IPNS name."),
 		cmdkit.BoolOption("nocache", "n", "Do not use cached entries."),
+		cmdkit.UintOption("dht-record-count", "dhtrc", "Number of records to request for DHT resolution."),
+		cmdkit.UintOption("dht-timeout", "dhtt", "Timeout in seconds for DHT resolution. Pass 0 for no timeout."),
```

Comment:
So, I might be missing something but IPFS commands *have* a timeout flag (e.g., `ipfs --timeout=30s`). And we can always set a DHT resolution timeout by setting a timeout on the context. What's the usecase for the additional timeout?

Reply:
The use case is that with --dht-timeout the DHT collects as many values as it can within the timeout, and then returns the best value it has found. The global IPFS command --timeout will simply kill the command:
```
> ./cmd/ipfs/ipfs name resolve --timeout=5s QmSgxPeqLrnM1ZB1pno3tYMQfqLB4cWXuAQNfSWsk77apK
Error: Post http://127.0.0.1:5001/api/v0/name/resolve?arg=QmSgxPeqLrnM1ZB1pno3tYMQfqLB4cWXuAQNfSWsk77apK&encoding=json&stream-channels=true&timeout=5s: context deadline exceeded
> ./cmd/ipfs/ipfs name resolve --dhtt=5 QmSgxPeqLrnM1ZB1pno3tYMQfqLB4cWXuAQNfSWsk77apK
/ipfs/QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn
```

Reply:
Maybe it would be better for the `--dht-timeout` parameter to take a string though, eg `--dht-timeout=5s` instead of assuming seconds eg `--dht-timeout=5`

Reply:
I see. It would be nice to explain that in the help but I can't think of a succinct way of putting it...

Reply:
@dirkmc it will be better for it to parse the time.

Reply:
I added a commit to parse the time, and changed the parameter description to
`Max time to collect values during DHT resolution eg \"30s\". Pass 0 for no timeout`
That's the most succinct way I can think of to say it